src/prng/isaac64.rs

```diff
@@ -118,28 +118,17 @@ impl Isaac64Rng {
     ///
     /// DEPRECATED. `Isaac64Rng::new_from_u64(0)` will produce identical results.
     #[deprecated(since="0.5.0", note="use the NewRng or SeedableRng trait")]
-    pub fn new_unseeded() -> Isaac64Rng {
+    pub fn new_unseeded() -> Self {
         Self::new_from_u64(0)
     }
 
-    /// Create an ISAAC-64 random number generator using an u64 as seed.
+    /// Create an ISAAC-64 random number generator using an `u64` as seed.
     /// If `seed == 0` this will produce the same stream of random numbers as
     /// the reference implementation when used unseeded.
-    pub fn new_from_u64(seed: u64) -> Isaac64Rng {
+    pub fn new_from_u64(seed: u64) -> Self {
         let mut key = [w(0); RAND_SIZE];
         key[0] = w(seed);
-        Isaac64Rng(BlockRng64 {
-            // Initialize with only one pass.
-            // A second pass does not improve the quality here, because all of
-            // the seed was already available in the first round.
-            // Not doing the second pass has the small advantage that if
-            // `seed == 0` this method produces exactly the same state as the
-            // reference implementation when used unseeded.
-            core: Isaac64Core::init(key, 1),
-            results: IsaacArray::default(),
-            index: RAND_SIZE, // generate on first use
-            half_used: false,
-        })
+        Isaac64Rng(BlockRng64::new(Isaac64Core::new_from_u64(seed)))
```

Comment:
You don't *need* the separate `Isaac64Core::new_from_u64` function, but if you do want it you don't need `key` on the line above.

Reply:
Sloppy work... I am sorry. Modified the commit.